examples/seeder_example/section_control_implement_sim.cpp

```diff
@@ -0,0 +1,452 @@
+					}
+				}
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualCountPerAreaApplicationRate):
+			{
+				value = sim->get_actual_rate();
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualWorkState):
+			{
+				value = sim->get_actual_work_state();
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::DeviceElementOffsetX):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::DeviceElementOffsetY):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::RequestDefaultProcessData):
+			{
+				value = 0;
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualWorkingWidth):
+			{
+				value = BOOM_WIDTH;
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState1_16):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState17_32):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState33_48):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState49_64):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState65_80):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState81_96):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState97_112):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState113_128):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState129_144):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState145_160):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState161_176):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState177_192):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState193_208):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState209_224):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState225_240):
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState241_256):
+			{
+				std::uint8_t sectionIndexOffset = NUMBER_SECTIONS_PER_CONDENSED_MESSAGE * (DDI - static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SetpointCondensedWorkState1_16));
+				value = 0;
+
+				for (std::uint_fast8_t i = 0; i < NUMBER_SECTIONS_PER_CONDENSED_MESSAGE; i++)
+				{
+					if ((i + sectionIndexOffset) < sim->get_number_of_sections())
+					{
+						value |= sim->get_section_setpoint_state(i + sectionIndexOffset) << (2 * i);
+					}
+					else
+					{
+						value |= (static_cast<std::uint32_t>(0x03) << (2 * i));
+					}
+				}
+			}
+			break;
```

Comment:
question: @ad3154 is setting the setpoint work state from the client the right thing to do? I have a feeling the server would be the only one determining the setpoint, but can't seem to find the details in the standard.

Reply:
Yeah, so it sounds like in general a TC is both entitled to request all settable values, and is likely to actually based on what I've been hearing. It seems like it would generally request them in case there's some other means of those values being manipulated outside of the TC connection, like manually setting the target rate from the implement's UT is a good example. Seems maybe a little odd for section states but apparently it's pretty common.